pkg/dbg/profile_dump.go

```diff
@@ -41,19 +62,10 @@ func DumpGoMemoryTrace() {
 
 // DumpGoProfile output goroutines to file.
 func DumpGoProfile() error {
-	trace := make([]byte, 5120*1024)
-	len := runtime.Stack(trace, true)
-	return ioutil.WriteFile(path+GetHostNamePrefix()+"-"+GetTimeStamp()+".stack", trace[:len], 0644)
+	return GoProfileDump(Path + GetHostNamePrefix() + "-" + GetTimeStamp() + ".stack")
 }
 
+// DumpHeap output heap to file.
 func DumpHeap() {
-	f, err := os.Create(path + GetHostNamePrefix() + "-" + GetTimeStamp() + ".heap")
-	if err != nil {
-		logrus.Errorf("could not create memory profile: %v", err)
-		return
-	}
-	defer f.Close()
-	if err := pprof.WriteHeapProfile(f); err != nil {
-		logrus.Errorf("could not write memory profile: %v", err)
-	}
+	HeapDump(Path + GetHostNamePrefix() + "-" + GetTimeStamp() + ".heap")
```

Comment:
These function names are a bit confusing now -- e.g. _"what's the difference between DumpGoProfile and GoProfileDump?"_, and _"how about HeapDump and DumpHeap?"_

I'd suggest to either
a. name them the same but change capitalization (e.g. `func DumpHeap()` vs. unexported `func dumpHeap(dest)`), or
b. if we want to keep them exported, perhaps have `func DumpGoProfile()` vs. `func DumpGoProfileTo()`

Reply:
There is no difference don't want to change DumpGoProfile its used in other places in the code.  The change was done to allow changes only to the profiles generated by stats thread in PX.  So wanted to add an exported layer which can be used with a specified filename.  That is all.     I will change the names to `GoProfileDump` to `DumpGoProfileTo` and `HeapDump` to `HeapDumpTo`.